comp/core/workloadmeta/collectors/internal/process/process_collector.go

```diff
@@ -0,0 +1,149 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+// Package process implements the local process collector for Workloadmeta.
+package process
+
+import (
+	"context"
+	"strconv"
+	"time"
+
+	"go.uber.org/fx"
+
+	workloadmeta "github.com/DataDog/datadog-agent/comp/core/workloadmeta/def"
+	"github.com/DataDog/datadog-agent/pkg/config"
+	"github.com/DataDog/datadog-agent/pkg/errors"
+	processwlm "github.com/DataDog/datadog-agent/pkg/process/metadata/workloadmeta"
+	"github.com/DataDog/datadog-agent/pkg/status/health"
+	"github.com/DataDog/datadog-agent/pkg/util/flavor"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+const (
+	collectorID   = "local-process-collector"
+	componentName = "workloadmeta-process"
+)
+
+type collector struct {
+	id      string
+	store   workloadmeta.Component
+	catalog workloadmeta.AgentType
+
+	processDiffCh <-chan *processwlm.ProcessCacheDiff
+}
+
+// NewCollector returns a new local process collector provider and an error
+func NewCollector() (workloadmeta.CollectorProvider, error) {
+	return workloadmeta.CollectorProvider{
+		Collector: &collector{
+			id:      collectorID,
+			catalog: workloadmeta.NodeAgent,
+		},
+	}, nil
+}
+
+// GetFxOptions returns the FX framework options for the collector
+func GetFxOptions() fx.Option {
+	return fx.Provide(NewCollector)
+}
+
+func (c *collector) enabled() bool {
+	if flavor.GetFlavor() != flavor.DefaultAgent {
+		return false
+	}
+
+	processChecksInCoreAgent := config.Datadog().GetBool("process_config.process_collection.enabled") &&
```

Comment:
why does live process collection need to be enabled? 

The process-agent version of the process collector doesn't rely on live process collection to enable the collector. It only relies on language detection. (Check [here](https://github.com/DataDog/datadog-agent/blob/201b789380d9c9faac6339d714aced6f0f4a45d8/pkg/process/metadata/workloadmeta/collector/process.go#L120-L133))

It might even cause some issues if a user has the following:
- language detection enabled
- live process collection enabled
- process config run in core agent set to true

I don't think there will be anything that stops having both the process collector and process check running simultaneously on the core agent.

Reply:
We discussed this off Github. This PR is meant to serve as an intermediary step. Its the easiest case to support language detection when the process check is enabled which is why it was tackled first. As a follow-up, we can work on adding the ability to use this without requiring the process check. Worth noting that even then, the remote process collector will still be required on Windows for the foreseeable future as process collection on Windows requires elevated privileges that the core agent does not have.

As for the case where both collectors start, leaving it as it is won't cause issues other than constant retries to begin the stream. To handle it, we can have a check like this:
```go
processChecksInCoreAgent := config.Datadog().GetBool("process_config.process_collection.enabled") &&
		config.Datadog().GetBool("process_config.run_in_core_agent.enabled")
langDetectionEnabled := config.Datadog().GetBool("language_detection.enabled")

return langDetectionEnabled && !processChecksInCoreAgent
```
It may worth having an OS check for Windows to help handle misconfigurations as well. What do you think?

Reply:
Thanks for clarifying.

> It may worth having an OS check for Windows to help handle misconfigurations as well (i.e. someone enables the run in core agent config on Windows). What do you think?

I think it depends on how you wish to expose the feature. 
Are we going to publicly document this and expose it via helm values.yaml for example? 
If yes, then I don't think we need to handle this misconfiguration option because we can simply indicate in the documentation that this option is not supported for windows. If a user tries it on windows and it doesn't work, then it is expected, based on the documentation. 

If no (i.e. if we need to enable/disable this option behind the scenes), then I think we should have an os check somewhere to make surely it is properly configured always.

WDYT?